students/prgrover/lesson04/mailroom.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+
+from collections import OrderedDict
+
+# Global dictionary containing donor list and donation amounts
+donor_dict = {"Harry Potter": [10000, 5000, 500.55], "Ronald Weasley": [2499.99, 7500.01], "Hermione Granger": [100, 2000, 30000], "Draco Malfoy": [10, 888.88], "Neville Longbottom": [10]}
+
+# Email template for Thank You message
+email = """Dear {:s},
+
+          Thank you for your very kind donation of ${:.2f}.
+          
+          It will be put to very good use.
+                         
+                Sincerely,
+                    -Hogwarts School of Witchcraft and Wizardry
+        """
+
+
+def send_thankyou():
+    """
+    Provides the following functionality:
+    1. Lists all donors
+    2. Adds new donors
+    3. Enters new donations
+    4. Sends a thank you email
+
+    Args: None
+    """
+    donor_name = input("Please enter the full name of a donor. You can also enter list to see all current donors or e to exit: ")
+    if donor_name == "e":
+        return
+
+    while donor_name != "list":
+        donation_amount = input("Please entergiftt the donation amount or e to exit: ")
+        if donation_amount == "e":
+            return
+        elif donor_name in donor_dict:
+            donation_amount = float(donation_amount)
+            donor_dict[donor_name].append(donation_amount)
+            print(email.format(donor_name, donation_amount))
+        else:
+            donation_amount = float(donation_amount)
+            donor_dict[donor_name] = [float(donation_amount)]
+            print(email.format(donor_name, donation_amount))
```

Comment:
You could eliminate some repetition here moving the shared functionality for new and existing donors outside of these conditional statements (float conversion and print statements)